src/main/java/com/piranframework/darbaan/PermissionCache.java

```diff
@@ -28,14 +28,15 @@
  * @author Isa Hekmatizadeh
  */
 class PermissionCache {
-  private Map<String, Collection<String>> permissions = new TreeMap<>();
+
+  private final Map<String, Collection<String>> permissions = new TreeMap<>();
 
   void addPermission(String actionAddress, Collection<String> roles) {
     permissions.put(actionAddress, roles);
   }
 
   boolean hasAccess(String serviceId, String actionCategory, String action, String role) {
-    Collection<String> roles = permissions.get(serviceId + "/" + actionCategory + "/" + action);
-    return !Objects.isNull(roles) && roles.contains(role);
+    Collection<String> roles = permissions.get(String.format("%s/%s/%s", serviceId, actionCategory, action));
+    return Objects.nonNull(roles) && !roles.isEmpty() && roles.contains(role);
```

Comment:
!roles.isEmpty() is redundant here as we check if roles contains the "role"

Reply:
You're right, going to fix.